rekall-core/rekall/plugins/linux/heap_analysis.py

```diff
@@ -0,0 +1,4388 @@
+            temp_vma['vma'] = vma
+            result.append(temp_vma)
+
+        return sorted(result, key=lambda vma: vma['vma'].vm_start)
+
+
+    def _load_libc_profile(self):
+        """Loads the Libc profile for the current libc version. """
+
+        try:
+            if self.__glibc_profile_filename:
+                libc_profile_file = open(self.__glibc_profile_filename, 'r').read()
+                libc_json_profile = json.loads(libc_profile_file)
+                self.session.logging.info("Loading libc profile: " + self.__glibc_profile_filename)
+
+                self._libc_profile = self.profile.LoadProfileFromData(
+                    libc_json_profile,
+                    name='libc',
+                    session=self.session)
+
+        except IOError as ex:
+            self.session.logging.error("Libc profile read error: %s", ex)
+
+        except Exception as ex:
+            self.session.logging.error("Fatal Error: %s", ex)
+            if self.session.GetParameter("debug"):
+                pdb.post_mortem()
+
+        if not self._libc_profile:
+            # we try to gather version information from the libc filename
+            libc_version = None
+            match = None
+            libc_filename = get_libc_filename(self.vmas)
+
+            if libc_filename:
+                match = re.search(r'(\d)\.(\d+)', libc_filename)
+
+            if match and len(match.groups()) == 2:
+                if match.group(1) == '2':
+                    if int(match.group(2)) >= 25:
+                        libc_version = '2.25'
+
+                    elif 23 <= int(match.group(2)) <= 24:
+                        libc_version = '2.23'
+
+                    else:
+                        libc_version = '2.20'
+
+            if not libc_version:
+                # Fallback
+                libc_version = '2.25'
+
+            self.session.logging.info("Loading profile for version {:s}.".format(libc_version))
+
+            if self.session.profile.metadata("arch") == 'I386':
+                self._libc_profile = GlibcProfile32(version=libc_version, session=self.session)
+
+            elif self.session.profile.metadata("arch") == 'AMD64':
+                self._libc_profile = GlibcProfile64(version=libc_version, session=self.session)
+
+
+        if not self._libc_profile:
+            self.session.logging.error('Unable to load a libc profile.')
+
+        else:
+            # merging profiles with self.profile.merge(self._libc_profile) leads to weird side
+            # effects (such as vm_area_struct names getting screwed up). Not sure at the moment
+            # why. Hence, we use those profiles separately, resp. merge them the other way around
+            self._libc_profile.merge(self.profile)
+
+            self._libc_profile.add_classes(dict(
+                malloc_state=malloc_state,
+                _heap_info=_heap_info,
+                malloc_chunk=malloc_chunk
+            ))
+
+
+    def next_chunk(self, chunk):
```

Comment:
Maybe its better to add an iterator interface to the malloc_chunk class so we just iterate over them?